### PR TITLE
[15.0][MIG] website_sale_vat_required

### PR DESCRIPTION
**Pull request by ChrisOForgeFlow**
Standard Migration

### CODE REVIEW

File: website_sale_vat_required/__manifest__.py [hunk 1]
```diff
@@ -5,13 +5,18 @@
...
     "author": "Agile Business Group, Tecnativa, Odoo Community Association (OCA)",
     "website": "https://github.com/OCA/e-commerce",
     "license": "AGPL-3",
     "depends": ["website_sale", "base_vat"],
-    "data": ["views/templates.xml", "views/assets.xml"],
+    "data": ["views/templates.xml"],
     "installable": True,
     "auto_install": False,
+    "assets": {
+        "web.assets_frontend": [
```
Comment (chienandalu) [inline website_sale_vat_required/__manifest__.py]:
```suggestion
        "web.assets_tests": [
```

**Comment by HaraldPanten** [comment]
Hi @ChrisOForgeFlow will you continue with the migration and attend comments?

Regards.

**Comment by ChrisOForgeFlow** [comment]
> Hi @ChrisOForgeFlow will you continue with the migration and attend comments?
> 
> Regards.

Sure! Done!

**Comment by HaraldPanten** [comment]
@ChrisOForgeFlow Could you check tests?

THX!

**Comment by cormaza** [comment]
> @ChrisOForgeFlow Could you check tests?
> 
> THX!

Done!, I change selector to make compatible with website_sale_checkout_country_vat

**Review by luis-ron — CHANGES_REQUESTED** [review]
@ChrisOForgeFlow  I have performed a series of functional tests and I have come across a case that may not be covered. 

The case is the following:

When accessing the web and making the purchase process, I find that, if I set as country United States, Vietnam or Brazil, I can proceed with the payment process even without setting a VAT in the field, when in principle, the use of this module is to make the field mandatory.

Could you please check it?

**Comment by cormaza** [comment]
> @ChrisOForgeFlow I have performed a series of functional tests and I have come across a case that may not be covered.
> 
> The case is the following:
> 
> When accessing the web and making the purchase process, I find that, if I set as country United States, Vietnam or Brazil, I can proceed with the payment process even without setting a VAT in the field, when in principle, the use of this module is to make the field mandatory.
> 
> Could you please check it?

I think this module works well alone without module website_sale_checkout_country_vat installed

https://github.com/OCA/e-commerce/blob/f3f6fad35dbeaa4ca63909c8db576f4c6f132983/website_sale_vat_required/controllers/main.py#L11

Because implementation of this module hide required field, and change the layout of view and set value of vat to complete_field, this one has always something because add prefix of country, because that for odoo is not empty vat field

https://github.com/OCA/e-commerce/blob/fc3982ab22083ee7fd2fc356d11adea283fa3046/website_sale_checkout_country_vat/views/templates.xml#L8-L10

Maybe somebody can add a third module that glue two implementations

**Review by HaraldPanten — APPROVED** [review]
Functional review. The module is working fine individually.

**Review by luis-ron — APPROVED** [review]
Functional review, LGTM

**Comment by HaraldPanten** [comment]
@pedrobaeza @chienandalu is that PR ready for merging?

THX!

**Review by chienandalu — APPROVED** [review]

**Pull request merged**